File: TaskManagerCLI/Task.cpp
#include "Task.h"
#include <sstream>

Task::Task(int id, const std::string& title, const std::string& description, const std::string& dueDate, TaskPriority priority)
    : id(id), title(title), description(description), dueDate(dueDate), priority(priority), completed(false) {}

int Task::getId() const {
    return id;
}

std::string Task::getTitle() const {
    return title;
}

std::string Task::getDescription() const {
    return description;
}

std::string Task::getDueDate() const {
    return dueDate;
}

TaskPriority Task::getPriority() const {
    return priority;
}

bool Task::isCompleted() const {
    return completed;
}

void Task::setTitle(const std::string& newTitle) {
    title = newTitle;
}

void Task::setDescription(const std::string& newDescription) {
    description = newDescription;
}

void Task::setDueDate(const std::string& newDueDate) {
    dueDate = newDueDate;
}

void Task::setPriority(TaskPriority newPriority) {
    priority = newPriority;
}

void Task::markComplete() {
    completed = true;
}

std::string Task::toString() const {
    std::stringstream ss;
    ss << "ID: " << id << "\n"
       << "Title: " << title << "\n"
       << "Description: " << description << "\n"
       << "Due Date: " << dueDate << "\n"
       << "Priority: ";
    switch (priority) {
        case TaskPriority::Low:
        case TaskPriority::Medium:
        case TaskPriority::High:
            ss << "High";
            break;
    }
    ss << "\nStatus: " << (completed ? "Completed" : "Pending") << "\n";
    return ss.str();
}

File: src/main/java/com/taskmanager/Task.java
package com.taskmanager;

import java.time.LocalDate;

public class Task {
    public enum TaskPriority {
        LOW,
        MEDIUM,
        HIGH
    }

    private int id;
    private String title;
    private String description;
    private LocalDate dueDate;
    private TaskPriority priority;
    private boolean completed;

    public Task(int id, String title, String description, LocalDate dueDate, TaskPriority priority) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.dueDate = dueDate;
        this.priority = priority;
        this.completed = false;
    }

    // Getters
    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public LocalDate getDueDate() {
        return dueDate;
    }

    public TaskPriority getPriority() {
        return priority;
    }

    public boolean isCompleted() {
        return completed;
    }

    // Setters
    public void setTitle(String newTitle) {
        this.title = newTitle;
    }

    public void setDescription(String newDescription) {
        this.description = newDescription;
    }

    public void setDueDate(LocalDate newDueDate) {
        this.dueDate = newDueDate;
    }

    public void setPriority(TaskPriority newPriority) {
        this.priority = newPriority;
    }

    public void markComplete() {
        this.completed = true;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("ID: ").append(id).append("\n")
          .append("Title: ").append(title).append("\n")
          .append("Description: ").append(description).append("\n")
          .append("Due Date: ").append(dueDate).append("\n")
          .append("Priority: ").append(priority).append("\n")
          .append("Status: ").append(completed ? "Completed" : "Pending").append("\n");
        return sb.toString();
    }
}